subsys/random/rand32_xoroshiro128.c

```diff
@@ -94,7 +95,7 @@ u32_t sys_rand32_get(void)
 {
 	u32_t ret;
 
-	if (k_sem_take(&state_sem, K_FOREVER) < 0) {
+	if (k_sem_take(&state_sem, K_NO_WAIT) < 0) {
```

Comment:
I don't understand the concern here? This is a semaphore over a simple block of code that is fairly fast even. Why would returning garbage instead of pseudorandom data ever be a better thing to do?

Reply:
BTW, if blocking ever becomes a concern, we would probably be better off coming up with a lock-free way of handling the generator state. Most systems with generators would have a state variable that a client could use (instead of a global). Lock-free with a 128-bit state would be tricky but there are ways of doing it.

The only place I see this being a real concern is if somehow we need pseudorandom values from an interrupt handler.